### PR TITLE
[NO GBP]Final Closet, Crate & Access Patches

### LINKED ISSUE #75313: Wrapping borked
## Testmerges: #73604
l
try to wrap crate with wrapping paper > you CANNOT.

this was on tram

### LINKED ISSUE #75349: Can't claim personal closets
## Reproduction:
slap a personal closet with your id
it does not unlock it and assign it to your id
shift click to examine it and see it isn't assigned to anyone

tried on 2 different personal closets so I'm PRETTY SURE it's related to claiming them.

### LINKED ISSUE #75351: Wallet don't seem to combine ID accesses
## Round ID:
Any recent round in the past few days

Put Assistant and Miner ID in wallet. While Assistant is in first place, Miner access doesn't work, and vice versa.

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request

1. Ensures the static list closet & crate paint job vars & the `access_choices` list vars are initialized only once. Since they are defined inside `Initialize()` they were initialized many times and need to be set only once when null

2. Fixes #75313
Closets & Crates can be wrapped again

3. Fixes #75349
Personal closets can be claimed again by anyone without ACCESS_ALL_PERSONAL_LOCKERS if it's currently unclaimed. 

4. Fixes #75351
Wallet with combined access are accepted again

**Note:** To claim a personal closet swipe once to unlock it, then swipe again to claim it

:cl:
fix: closets & crates can be wrapped with wrapping paper again
fix: personal closets can be claimed again by players without personal access level
fix: wallets with combined access work again
refactor: closet & crate paint jobs static list vars, access_choices static list var are initialized only once during init
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
@Time-Green review & merge?

File: code/modules/jobs/access.dm [hunk 1]
```diff
@@ -32,6 +32,12 @@
 	//if they are carying a card that has access, that works
 	else if(isliving(accessor))
 		var/mob/living/being = accessor
+		//humans can carry wallets that combine accesses so check for those as well.
+		if(ishuman(accessor))
+			var/mob/living/carbon/human/human_accessor = being
+			if(check_access(human_accessor.wear_id))
+				return TRUE
+		//check for id carried in hands, inside PDA's etc.
```
Comment (timothymtorres) [inline code/modules/jobs/access.dm:40]:
```suggestion
	//if they are wearing a card that has access, that works
	else if(ishuman(accessor))
		var/mob/living/carbon/human/human_accessor = accessor
		if(check_access(human_accessor.wear_id))
			return TRUE
	//if they have a hacky abstract animal ID with the required access, let them in i guess...
	else if(isanimal(accessor))
		var/mob/living/simple_animal/animal = accessor
		if(check_access(animal.access_card))
```

Do we want to keep the old code from before?  Most of this code was for simple bots like medibots to have paramedic access.  Can you double check that it still works correctly?

Comment (SyncIt21) [reply code/modules/jobs/access.dm:40]: Oh wait, i just looked at the code and i realized the case is covered for when the player is carrying the ID in their active hand
Screenshot (206)
Yup your suggestion can be applied, never mind

**Comment by Time-Green** [comment]
Very nice!

**Comment by SyncIt21** [comment]
Ahh shit, forgot to update the change log about the personal closets & wallet. Ahh well, it is what it is

**Pull request merged**